src/Microsoft.PowerShell.Commands.Utility/commands/utility/CsvCommands.cs

```diff
@@ -1044,7 +1044,15 @@ internal string ConvertPSObjectToCSV(PSObject mshObject, IList<string> propertyN
                 {
                     if (dictionary.Contains(propertyName))
                     {
-                        value = dictionary[propertyName].ToString();
+                        //Handle null reference exception for existing property with null value
+                        if (dictionary[propertyName] == null)
+                        {
+                            value = null;
+                        }
+                        else
+                        {
+                            value = dictionary[propertyName].ToString();
+                        }
```

Comment:
Can we simplify the code?
```suggestion
                        value = dictionary[propertyName]?.ToString();
```
Also please remove obvious comment.